c++ training/PointersTraining/PointerExercises/S2E4/Source.cpp

```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+
+/* Exercise 4 - Dynamic Allocation Dangers: Dangling Pointers
+There is another danger when using dynamic memory. Let�s see it:
+Allocate an int in the heap and assign it to an int pointer variable. Immediately after that, delete
+the allocated memory.
+We�ve previously discussed pointer validity. Check the validity of the pointer. If the pointer is
+valid, print �The pointer is valid!�. If it�s not, print �The pointer is not valid��.
+Run your program. You�ll see that the pointer is valid. Try accessing the value of whatever the
+pointer is pointing to and you�ll most likely crash. You�ve got a false positive in your validation
+due to a dangling pointer.
+Print to the console why you think this is happening. It�s important to keep in mind that after
+freeing memory in the heap, any pointer that was pointing to it should be set to null. From now
+on, this will be expected in the exercises.*/
+
+
+int main()
+{
+	int* SoonToBeADanglingPointer = new int(1);
+	delete SoonToBeADanglingPointer;
+
+	SoonToBeADanglingPointer = nullptr;
+
+	if (SoonToBeADanglingPointer)
+	{
+		std::cout << *SoonToBeADanglingPointer;
```

Comment:
I know that you print Soon To Be A Dangling Pointer, but it is not clear to me, it is not the expected validation, also in case the validation is not true, you are leaving it to the air and the idea is that you can try or test what happens with the exercise

Reply:
What do you mean it's not the expected validation?